Python 3.10 asyncio-related test failures

- [x] I have marked all applicable categories:
    + [x] exception-raising bug
    + [ ] visual output bug
- [x] I have visited the [source website], and in particular
  read the [known issues]
- [x] I have searched through the [issue tracker] for duplicates
- [x] I have mentioned version numbers, operating system and
  environment, where applicable:
  ```python
  import tqdm, sys
  print(tqdm.__version__, sys.version, sys.platform)
  ```

[source website]: https://github.com/tqdm/tqdm/
[known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
[issue tracker]: https://github.com/tqdm/tqdm/issues?q=

--------------
```
>>> print(tqdm.__version__, sys.version, sys.platform)
4.61.0 3.10.0b2 (default, Jun  1 2021, 09:14:18) [GCC 11.1.0] linux
```

This is git master (05ad200df3ccbc21c5c840d064e619ec9838f902). I've get three test failures. The first two are clearly related to the `loop` parameter being removed. The third one is above my pay grade ;-).

```pytb
_____________________________________________________________ test_as_completed[0.1] _____________________________________________________________
tests/py37_asyncio.py:106: in test_as_completed
    for i in as_completed([asyncio.sleep(0.01 * i) for i in range(30, 0, -1)]):
tqdm/asyncio.py:63: in as_completed
    yield from cls(asyncio.as_completed(fs, loop=loop, timeout=timeout),
E   TypeError: as_completed() got an unexpected keyword argument 'loop'
__________________________________________________________________ test_gather ___________________________________________________________________
tests/py37_asyncio.py:125: in test_gather
    res = await gather(list(map(double, range(30))))
tqdm/asyncio.py:75: in gather
    res = [await f for f in cls.as_completed(ifs, loop=loop, timeout=timeout,
tqdm/asyncio.py:-1: in <listcomp>
    ???
tqdm/asyncio.py:63: in as_completed
    yield from cls(asyncio.as_completed(fs, loop=loop, timeout=timeout),
E   TypeError: as_completed() got an unexpected keyword argument 'loop'
___________________________________________________________________ test_main ____________________________________________________________________
../venv/lib/python3.10/site-packages/_pytest/runner.py:340: in from_call
    result: Optional[TResult] = func()
../venv/lib/python3.10/site-packages/_pytest/runner.py:261: in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
../venv/lib/python3.10/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../venv/lib/python3.10/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../venv/lib/python3.10/site-packages/pluggy/manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../venv/lib/python3.10/site-packages/_pytest/unraisableexception.py:88: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
../venv/lib/python3.10/site-packages/_pytest/unraisableexception.py:78: in unraisable_exception_runtest_hook
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object double at 0x7ff659797bc0>
E   
E   Traceback (most recent call last):
E     File "/usr/lib/python3.10/warnings.py", line 506, in _warn_unawaited_coroutine
E       warn(msg, category=RuntimeWarning, stacklevel=2, source=coro)
E   RuntimeWarning: coroutine 'double' was never awaited

```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -164,12 +164,11 @@ jobs:
         fi
         git log --pretty='format:%d%n- %s%n%b---' $(git tag --sort=v:refname | tail -n2 | head -n1)..HEAD > _CHANGES.md
     - if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
-      id: create_release
       uses: softprops/action-gh-release@v1
       env:
         GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
       with:
-        release_name: tqdm ${{ github.ref }} stable
+        name: tqdm ${{ github.ref }} stable
         body_path: _CHANGES.md
         draft: true
         files: |

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -45,6 +45,7 @@ jobs:
         os: [macos-latest, windows-latest]
     name: py${{ matrix.python }}-${{ matrix.os }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
     steps:
     - uses: actions/checkout@v2
       with:
@@ -77,6 +78,7 @@ jobs:
         python: [2.7, 3.5, 3.6, 3.7, 3.8, 3.9]
     name: py${{ matrix.python }}
     runs-on: ubuntu-latest
+    timeout-minutes: 20
     steps:
     - uses: actions/checkout@v2
       with:

--- a/.meta/.readme.rst
+++ b/.meta/.readme.rst
@@ -588,9 +588,6 @@ Nested progress bars
             for k in trange(50, desc='3rd loop', leave=False):
                 sleep(0.01)
 
-On Windows `colorama <https://github.com/tartley/colorama>`__ will be used if
-available to keep nested bars on their respective lines.
-
 For manual control over positioning (e.g. for multi-processing use),
 you may specify ``position=n`` where ``n=0`` for the outermost bar,
 ``n=1`` for the next, and so on.

--- a/.meta/.readme.rst
+++ b/.meta/.readme.rst
@@ -1251,7 +1251,7 @@ Citation information: |DOI|
 .. |GitHub-Updated| image:: https://img.shields.io/github/last-commit/tqdm/tqdm/master.svg?logo=github&logoColor=white&label=pushed
    :target: https://github.com/tqdm/tqdm/pulse
 .. |Gift-Casper| image:: https://img.shields.io/badge/dynamic/json.svg?color=ff69b4&label=gifts%20received&prefix=%C2%A3&query=%24..sum&url=https%3A%2F%2Fcaspersci.uk.to%2Fgifts.json
-   :target: https://www.cdcl.ml/sponsor
+   :target: https://cdcl.ml/sponsor
 .. |Versions| image:: https://img.shields.io/pypi/v/tqdm.svg
    :target: https://tqdm.github.io/releases
 .. |PyPI-Downloads| image:: https://img.shields.io/pypi/dm/tqdm.svg?label=pypi%20downloads&logo=PyPI&logoColor=white

--- a/DEMO.ipynb
+++ b/DEMO.ipynb
@@ -58,7 +58,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "![Screenshot](https://img.tqdm.ml/tqdm.gif)|[![Video](https://img.tqdm.ml/video.jpg)](https://tqdm.github.io/video) [![Slides](https://img.tqdm.ml/slides.jpg)](https://tqdm.github.io/PyData2019/slides.html)\n",
+    "![Screenshot](https://img.tqdm.ml/tqdm.gif)|[![Video](https://img.tqdm.ml/video.jpg)](https://tqdm.github.io/video) [![Slides](https://img.tqdm.ml/slides.jpg)](https://tqdm.github.io/PyData2019/slides.html) [![Merch](https://img.tqdm.ml/merch.jpg)](https://tqdm.github.io/merch)\n",
     "-|-\n",
     "\n",
     "It can also be executed as a module with pipes:"
@@ -273,7 +273,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "! find . -name '*.py' -type f -exec cat \\{} \\; | tqdm --unit loc --unit_scale --total 4104300 >> /dev/null"
+    "! find . -name '*.py' -type f -exec cat \\{} \\; | tqdm --unit loc --unit-scale --total 4104300 --null"
    ]
   },
   {
@@ -699,9 +699,9 @@
     "\n",
     "### Keras Integration\n",
     "\n",
-    "```python\n",
     "A `keras` callback is also available:\n",
     "\n",
+    "```python\n",
     "from tqdm.keras import TqdmCallback\n",
     "\n",
     "...\n",

--- a/DEMO.ipynb
+++ b/DEMO.ipynb
@@ -472,9 +472,6 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "On Windows [colorama](https://github.com/tartley/colorama) will be used\n",
-    "if available to keep nested bars on their respective lines.\n",
-    "\n",
     "For manual control over positioning (e.g. for multi-processing use),\n",
     "you may specify `position=n` where `n=0` for the outermost bar, `n=1`\n",
     "for the next, and so on. However, it's best to check if tqdm can work\n",

--- a/README.rst
+++ b/README.rst
@@ -807,9 +807,6 @@ Nested progress bars
             for k in trange(50, desc='3rd loop', leave=False):
                 sleep(0.01)
 
-On Windows `colorama <https://github.com/tartley/colorama>`__ will be used if
-available to keep nested bars on their respective lines.
-
 For manual control over positioning (e.g. for multi-processing use),
 you may specify ``position=n`` where ``n=0`` for the outermost bar,
 ``n=1`` for the next, and so on.

--- a/README.rst
+++ b/README.rst
@@ -1470,7 +1470,7 @@ Citation information: |DOI|
 .. |GitHub-Updated| image:: https://img.shields.io/github/last-commit/tqdm/tqdm/master.svg?logo=github&logoColor=white&label=pushed
    :target: https://github.com/tqdm/tqdm/pulse
 .. |Gift-Casper| image:: https://img.shields.io/badge/dynamic/json.svg?color=ff69b4&label=gifts%20received&prefix=%C2%A3&query=%24..sum&url=https%3A%2F%2Fcaspersci.uk.to%2Fgifts.json
-   :target: https://www.cdcl.ml/sponsor
+   :target: https://cdcl.ml/sponsor
 .. |Versions| image:: https://img.shields.io/pypi/v/tqdm.svg
    :target: https://tqdm.github.io/releases
 .. |PyPI-Downloads| image:: https://img.shields.io/pypi/dm/tqdm.svg?label=pypi%20downloads&logo=PyPI&logoColor=white

--- a/setup.cfg
+++ b/setup.cfg
@@ -75,6 +75,8 @@ classifiers=
 [options]
 setup_requires=setuptools>=42; setuptools_scm[toml]>=3.4
 python_requires=>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*
+install_requires=
+    colorama; platform_system == 'Windows'
 tests_require=tox
 include_package_data=True
 packages=find:

--- a/tqdm/asyncio.py
+++ b/tqdm/asyncio.py
@@ -8,6 +8,7 @@ Usage:
 ...     ...
 """
 import asyncio
+from sys import version_info
 
 from .std import tqdm as std_tqdm
 
@@ -60,7 +61,11 @@ class tqdm_asyncio(std_tqdm):
         """
         if total is None:
             total = len(fs)
-        yield from cls(asyncio.as_completed(fs, loop=loop, timeout=timeout),
+
+        kwargs = {}
+        if version_info[:2] < (3, 10):
+            kwargs['loop'] = loop
+        yield from cls(asyncio.as_completed(fs, timeout=timeout, **kwargs),
                        total=total, **tqdm_kwargs)
 
     @classmethod

--- a/tqdm/asyncio.py
+++ b/tqdm/asyncio.py
@@ -61,7 +61,6 @@ class tqdm_asyncio(std_tqdm):
         """
         if total is None:
             total = len(fs)
-
         kwargs = {}
         if version_info[:2] < (3, 10):
             kwargs['loop'] = loop

--- a/tqdm/contrib/telegram.py
+++ b/tqdm/contrib/telegram.py
@@ -62,6 +62,17 @@ class TelegramIO(MonoWorker):
         else:
             return future
 
+    def delete(self):
+        """Deletes internal `message_id`."""
+        try:
+            future = self.submit(
+                self.session.post, self.API + '%s/deleteMessage' % self.token,
+                data={'chat_id': self.chat_id, 'message_id': self.message_id})
+        except Exception as e:
+            tqdm_auto.write(str(e))
+        else:
+            return future
+
 
 class tqdm_telegram(tqdm_auto):
     """
@@ -111,6 +122,13 @@ class tqdm_telegram(tqdm_auto):
         super(tqdm_telegram, self).clear(*args, **kwargs)
         if not self.disable:
             self.tgio.write("")
+
+    def close(self):
+        if self.disable:
+            return
+        super(tqdm_telegram, self).close()
+        if not (self.leave or (self.leave is None and self.pos == 0)):
+            self.tgio.delete()
 
 
 def ttgrange(*args, **kwargs):

--- a/tqdm/keras.py
+++ b/tqdm/keras.py
@@ -7,7 +7,7 @@ from .auto import tqdm as tqdm_auto
 
 try:
     import keras
-except ImportError as e:
+except (ImportError, AttributeError) as e:
     try:
         from tensorflow import keras
     except ImportError:

--- a/tqdm/rich.py
+++ b/tqdm/rich.py
@@ -113,10 +113,10 @@ class tqdm_rich(std_tqdm):  # pragma: no cover
         self._prog.__enter__()
         self._task_id = self._prog.add_task(self.desc or "", **d)
 
-    def close(self, *args, **kwargs):
+    def close(self):
         if self.disable:
             return
-        super(tqdm_rich, self).close(*args, **kwargs)
+        super(tqdm_rich, self).close()
         self._prog.__exit__(None, None, None)
 
     def clear(self, *_, **__):

--- a/tqdm/std.py
+++ b/tqdm/std.py
@@ -791,8 +791,12 @@ class tqdm(Comparable):
                         " Use keyword arguments instead.",
                         fp_write=getattr(t.fp, 'write', sys.stderr.write))
 
+                try:  # pandas>=1.3.0
+                    from pandas.core.common import is_builtin_func
+                except ImportError:
+                    is_builtin_func = df._is_builtin_func
                 try:
-                    func = df._is_builtin_func(func)
+                    func = is_builtin_func(func)
                 except TypeError:
                     pass
 

--- a/tqdm/std.py
+++ b/tqdm/std.py
@@ -336,6 +336,7 @@ class tqdm(Comparable):
         """
         fp = file
         fp_flush = getattr(fp, 'flush', lambda: None)  # pragma: no cover
+        fp_flush()
 
         def fp_write(s):
             fp.write(_unicode(s))

--- a/tqdm/std.py
+++ b/tqdm/std.py
@@ -336,7 +336,9 @@ class tqdm(Comparable):
         """
         fp = file
         fp_flush = getattr(fp, 'flush', lambda: None)  # pragma: no cover
-        fp_flush()
+        if fp in (sys.stderr, sys.stdout):
+            sys.stderr.flush()
+            sys.stdout.flush()
 
         def fp_write(s):
             fp.write(_unicode(s))

